src/main/java/org/embl/mobie/io/ome/zarr/util/ZarrArrayCreator.java

```diff
@@ -29,28 +29,14 @@ public A createArray(DataBlock<?> dataBlock, long[] gridPosition) {
 
     @Override
     public long[] getCellDims(long[] gridPosition) {
-        long[] cellMin = new long[3];
-        int[] cellDims = new int[3];
-
-        // TODO: do something like in: private long[] toZarrChunkIndices( long[] gridPosition )
-        if (zarrAxes.is4DWithChannels() || zarrAxes.is4DWithTimepoints()) {
-            cellMin = new long[4];
-            cellDims = new int[4];
-            cellDims[3] = 1; // channel
-        }
+        long[] cellMin = new long[Math.max(zarrAxes.getNumDimension(), 3)];
```

Comment:
Here, I am confused, because I forget how the cellGrid is initialized.
1. Is that representing the full zarray (all dimensions) or only the spatial part of it?
2. If the spatial part on the zarray side is only 2D, do we add a singleton dimension to the cellGrid? And thus you have the hard-coded 3?